root/types/edit_types.js

```diff
@@ -0,0 +1,1095 @@
+/*
+ * @flow strict
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+declare type AddAnnotationEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +changelog: string,
+    +entity_type: AnnotatedEntityTypeT,
+    [annotatedEntityType: AnnotatedEntityTypeT]: AnnotatedEntityT,
+    +html: string,
+    +text: string,
+  },
+  +edit_type: EDIT_ANNOTATION_ADD_T,
+}>;
+
+declare type AddAreaEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    ...DatePeriodRoleT,
+    +area: AreaT,
+    +comment: string | null,
+    +iso_3166_1: $ReadOnlyArray<string>,
+    +iso_3166_2: $ReadOnlyArray<string>,
+    +iso_3166_3: $ReadOnlyArray<string>,
+    +name: string,
+    +sort_name: string | null,
+    +type: AreaTypeT | null,
+  },
+  +edit_type: EDIT_AREA_CREATE_T,
+}>;
+
+declare type AddArtistEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    ...CommentRoleT,
+    ...DatePeriodRoleT,
+    +area: AreaT | null,
+    +artist: ArtistT,
+    +begin_area: AreaT | null,
+    +end_area: AreaT | null,
+    +gender: GenderT | null,
+    +ipi_codes: $ReadOnlyArray<string> | null,
+    +isni_codes: $ReadOnlyArray<string> | null,
+    +name: string,
+    +sort_name: string,
+    +type: ArtistTypeT | null,
+  },
+  +edit_type: EDIT_ARTIST_CREATE_T,
+}>;
+
+declare type AddCoverArtEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +artwork: ArtworkT,
+    +position: number,
+    +release: ReleaseT,
+  },
+  +edit_type: EDIT_RELEASE_ADD_COVER_ART_T,
+}>;
+
+declare type AddDiscIdEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +medium?: MediumT,
+    +medium_cdtoc: MediumCDTocT,
+  },
+  +edit_type: EDIT_MEDIUM_ADD_DISCID_T,
+}>;
+
+declare type AddEventEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    ...CommentRoleT,
+    ...DatePeriodRoleT,
+    +cancelled: boolean,
+    +ended: boolean,
+    +event: EventT,
+    +name: string,
+    +setlist: string,
+    +time: string | null,
+    +type: EventTypeT | null,
+  },
+  +edit_type: EDIT_EVENT_CREATE_T,
+}>;
+
+declare type AddInstrumentEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    ...DatePeriodRoleT,
+    +comment: string | null,
+    +description: string | null,
+    +instrument: InstrumentT,
+    +name: string,
+    +type: InstrumentTypeT | null,
+  },
+  +edit_type: EDIT_INSTRUMENT_CREATE_T,
+}>;
+
+declare type AddIsrcsEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +additions: $ReadOnlyArray<{
```

Comment:
`Cannot create 'EditNote' element because in property 'edit': Either property 'additions' is missing in  object type [1] but exists in  object type [2] in property 'display_data'. [etc]`

Not sure why this one is a problem TBH?